### DISCUSSION TITLE
Unexpected SysEx Behavior

### DISCUSSION THREAD

**Discussion by randel0** [FortySevenEffects/arduino_midi_library#198] [Q&A] [answered]
This is also posted on the Arduino forum here.
Hello-
I'm seeing some unexpected behavior when using the MIDI library to parse a particular string of System Exclusive.
Nearly all SysEx that I send works as expected, except for a patch dump from a Sequential Prophet VS synthesizer.
Here is my test code running on a Adafruit ItsyBitsy and version 5.0.2 of the MIDI library:

```cpp
#include <MIDI.h>
MIDI_CREATE_DEFAULT_INSTANCE();
// -----------------------------------------------------------------------------
// This example shows the old way of checking for input messages.
// It's simpler to use the callbacks now, check out the dedicated example.
#define LED 13                   // LED pin on Arduino Uno
// -----------------------------------------------------------------------------
void BlinkLed(byte num)         // Basic blink function
{
for (byte i=0;i<num;i++)
{
digitalWrite(LED,HIGH);
delay(50);
digitalWrite(LED,LOW);
delay(50);
}
}
// -----------------------------------------------------------------------------
void setup()
{
pinMode(LED, OUTPUT);
MIDI.begin();           // Launch MIDI, by default listening to channel 1.
}
void loop()
{
if (MIDI.read())                // Is there a MIDI message incoming ?
{
switch(MIDI.getType())      // Get the type of the message we caught
{
case midi::SystemExclusive:       // If it is SysEx,
BlinkLed(5);                // blink the LED 5 times
break;
// See the online reference for other message types
default:
break;
}
}
```

If the microcontroller recognizes incoming SysEx data, it blinks the LED 5 times.
This works as expected for a variety of synthesizers.
Here is the SysEx string that breaks it:

```
00  F0 01 0A 00 09 07 02 0E  05 0C 0B 00 00 00 00 00  |                |
10  00 00 02 0C 01 09 01 04  0E 00 00 0C 08 01 01 02  |                |
20  01 01 00 02 00 06 0A 00  05 09 00 01 08 0E 0A 0D  |                |
30  00 00 01 05 0B 0C 06 0B  05 06 04 0B 02 0C 06 0D  |                |
40  01 06 00 00 00 00 05 0A  04 05 04 05 05 0C 08 00  |                |
50  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |                |
60  00 00 0F 06 0A 08 00 00  00 00 00 02 0E 08 0E 0F  |                |
70  0C 02 00 09 0F 0B 0D 0F  0D 07 0E 01 09 0C 0E 0A  |                |
80  0E 01 00 01 07 0F 0D 07  0C 00 02 00 00 0C 00 0E  |                |
90  02 00 02 04 04 04 06 04  00 00 00 03 00 03 00 01  |                |
A0  04 00 02 06 04 06 00 00  F7
```

To decode the first few bytes:
F0 - Start of SysEx
01 - Manufacturer's ID, Sequential/DSI
0A - Specific device, Prophet VS
00...00 - Data
F7 - End of SysEx
I'm stumped.
Is there something in the MIDI Library configuration file that needs adjusting? Is something weird with the Sequential checksum scheme?
Any input would be appreciated. Thanks!

**Comment by franky47** [comment, marked as answer]
It looks like a long message, by default the library handles SysEx up to 128 bytes.

Any message longer than that may cause issues.

Fortunately, you can increase the maximum SysEx size in the [settings](https://github.com/FortySevenEffects/arduino_midi_library/wiki/Using-custom-Settings#overriding-settings):

```cpp
struct MySettings : public midi::DefaultSettings
{
    static const unsigned SysExMaxSize = 256; // Accept SysEx messages up to 256 bytes long.
};

// Create a 'MIDI' object using MySettings bound to Serial2.
MIDI_CREATE_CUSTOM_INSTANCE(HardwareSerial, Serial2, MIDI, MySettings);
```

**Reply by randel0** [reply]
franky47-

Thank you so much for your swift reply, which solved my problem immediately.

Also, thank you for your devotion to the development and upkeep of this library, and effectively communicating to the community - you are making the world a better place!

**Reply by franky47** [reply]
Thanks for the kind words, much appreciated!
